archive/miu_language/arithmetic.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2020 Gihan Marasingha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gihan Marasingha
+-/
+import data.nat.modeq
+import tactic.linarith
+
+/-!
+# Arithmetic
+
+A basic arithmetic result needed to analyse the MIU system.
+-/
+
+open nat
+
+private lemma le_pow2_and_pow2_eq_mod3' (c : ℕ) (x : ℕ) (h : c = 1 ∨ c = 2) :
+  ∃ m : ℕ, c + 3*x ≤ 2^m ∧ 2^m % 3 = c % 3 :=
```

Comment:
Do you want the `c % 3` at the end of the statement? You might just as well write `c`, since `c = 1` or `c = 2`.

Reply:
True, but the current statement of this lemma makes unification in its corollary, `le_pow2_and_pow2_eq_mod3`, easier. I'd rather keep this as it is.

Reply:
Sure... fine with me.